src/client/OpenFin/provider.ts

```diff
@@ -1,29 +1,217 @@
 import OpenFin from "@openfin/core"
+import { ENVIRONMENT } from "client/constants"
+
+type ExternalClientMap = Map<
+  OpenFin.ApplicationIdentity["uuid"],
+  OpenFin.InteropClient
+>
+
+export interface ExternalContext extends OpenFin.Context {
+  _clientInfo?: {
+    uuid: string
+  }
+}
+
+function interopOverride(
+  InteropBroker: OpenFin.Constructor<OpenFin.InteropBroker>,
+): OpenFin.InteropBroker {
+  class Override extends InteropBroker {
+    public analyticsUuid: string
+
+    public limitCheckerUuid: string
+
+    public analyticsClients: ExternalClientMap
+
+    public limitCheckerClients: ExternalClientMap
+
+    constructor() {
+      super()
+      this.analyticsUuid = `reactive-analytics-${ENVIRONMENT}`
+      this.limitCheckerUuid = `reactive-trader-limit-checker-${ENVIRONMENT}`
+      this.analyticsClients = new Map()
+      this.limitCheckerClients = new Map()
+      this.initializeBrokers(this.analyticsUuid, this.analyticsClients).catch(
+        (error) => console.error(error),
+      )
+      this.initializeBrokers(
+        this.limitCheckerUuid,
+        this.limitCheckerClients,
+      ).catch((error) => console.error(error))
+    }
+
+    public async initializeBrokers(
+      externalUuid: string,
+      externalClients: ExternalClientMap,
+    ): Promise<void> {
+      const platform: OpenFin.Platform = fin.Platform.wrapSync({
+        uuid: externalUuid,
+      })
+
+      // If platform is already running
+      if (await platform.Application.isRunning()) {
+        await this.setupContextGroups(externalUuid, externalClients)
+      }
+
+      // If platform is just being run
+      await platform.on("platform-api-ready", async () => {
+        await this.setupContextGroups(externalUuid, externalClients)
+      })
+
+      await platform.Application.once("closed", () => {
+        externalClients = new Map()
+      })
+    }
+
+    public async setupContextGroups(
+      externalUuid: string,
+      externalClients: ExternalClientMap,
+    ) {
+      // Create a InteropClient instance by connecting to the external broker
+      const externalInteropClient = fin.Interop.connectSync(externalUuid, {})
+      const externalContextGroup =
+        await externalInteropClient.getContextGroups()
+
+      // Create a InteropClient instance by connecting to the current platform broker
+      const fxInteropClient = fin.Interop.connectSync(fin.me.uuid, {})
```

Comment:
minor: probably doesn't matter for this pass, but can we remove the fx-specificity (or is this not generic enough for Credit yet?)

Reply:
Renamed